File: app/src/main/java/com/android3/siegertpclient/data/invitation/invitationsource/invitationLocal/InvitationLocalDataSource.java
package com.android3.siegertpclient.data.invitation.invitationsource.invitationLocal;

import com.android3.siegertpclient.data.invitation.Invitation;
import com.android3.siegertpclient.data.setting.Setting;
import com.android3.siegertpclient.data.setting.settingsource.settingLocal.SettingDao;

import org.jetbrains.annotations.NotNull;

public class InvitationLocalDataSource {

        final InvitationDao invitationDao;

    public InvitationLocalDataSource(InvitationDao invitationDao) {
        this.invitationDao = invitationDao;
    }

        public void saveInvitation(@NotNull Invitation newInvitation) {
            invitationDao.insertInvitation(newInvitation);
        }

    }

File: app/src/main/java/com/android3/siegertpclient/data/invitation/invitationsource/invitationLocal/InvitationLocalDataSource.kt
package com.android3.siegertpclient.data.invitation.invitationsource.invitationLocal

import com.android3.siegertpclient.data.Result.Result
import com.android3.siegertpclient.data.invitation.Invitation

class InvitationLocalDataSource(val invitationDao: InvitationDao) {
    fun saveInvitation(newInvitation: Invitation) {
        invitationDao.insertInvitation(newInvitation)
    }
    fun getInvitation(){
        invitationDao.allInvitations
    }
    fun deleteResult(){
        invitationDao.deleteAllInvitations()
    }
    fun updateResult(invitation: Invitation){
        invitationDao.updateInvitation(invitation)
    }
}